libraries/AP_IBus_Telem/AP_IBus_Telem.cpp

```diff
@@ -0,0 +1,796 @@
+        break;
+
+#if AP_AHRS_ENABLED
+    case IBUS_SENSOR_TYPE_COMPASS_HEADING:
+        value.uint16 = AP::ahrs().yaw_sensor * 0.01;
+        break;
+#endif
+
+#if AP_BARO_ENABLED
+    case IBUS_SENSOR_TYPE_CLIMB_RATE:
+    case IBUS_SENSOR_TYPE_VERTICAL_SPEED:
+        value.int16 = AP::baro().get_climb_rate() * 100;
+        break;
+#endif
+
+#if AP_GPS_ENABLED
+    case IBUS_SENSOR_TYPE_COG:
+        value.uint16 = AP::gps().ground_course() * 100;
+        break;
+
+    case IBUS_SENSOR_TYPE_GPS_STATUS:
+        value.byte[0] = get_gps_status();
+        value.byte[1] = AP::gps().num_sats();
+        break;
+#endif // AP_GPS_ENABLED
+
+#if AP_AHRS_ENABLED
+    case IBUS_SENSOR_TYPE_ACC_X:
+        value.int16 = (AP::ahrs().get_accel() - AP::ahrs().get_accel_bias()).x * 1000;
+        break;
+
+    case IBUS_SENSOR_TYPE_ACC_Y:
+        value.int16 = (AP::ahrs().get_accel() - AP::ahrs().get_accel_bias()).y * 1000;
+        break;
+
+    case IBUS_SENSOR_TYPE_ACC_Z:
+        value.int16 = (AP::ahrs().get_accel() - AP::ahrs().get_accel_bias()).z * 1000;
+        break;
+
+    case IBUS_SENSOR_TYPE_ROLL:
+        value.int16 = AP::ahrs().roll_sensor;
+        break;
+
+    case IBUS_SENSOR_TYPE_PITCH:
+        value.int16 = AP::ahrs().pitch_sensor;
+        break;
+
+    case IBUS_SENSOR_TYPE_YAW:
+        value.int16 = AP::ahrs().yaw_sensor;
+        break;
+#endif // AP_AHRS_ENABLED
+
+#if AP_GPS_ENABLED
+    case IBUS_SENSOR_TYPE_GROUND_SPEED:
+        value.uint16 = AP::gps().ground_speed_cm();
+        break;
+#endif
+
+    case IBUS_SENSOR_TYPE_GPS_DIST:
+        value.uint16 = get_distance_from_home_m();
+        break;
+
+#if AP_ARMING_ENABLED
+    case IBUS_SENSOR_TYPE_ARMED:
+        value.uint16 = AP::arming().is_armed();
+        break;
+#endif
+
+    case IBUS_SENSOR_TYPE_FLIGHT_MODE:
+        value.uint16 = get_vehicle_mode();
+        break;
+
+#if AP_AIRSPEED_ENABLED
+    case IBUS_SENSOR_TYPE_SPEED:
+        value.uint16 = AP::airspeed()->get_airspeed();
+        break;
+#endif
+
+#if AP_BARO_ENABLED
+    case IBUS_SENSOR_TYPE_TEMPERATURE_PRESSURE: {
+        const uint32_t pressure = AP::baro().get_pressure();
+        const uint32_t temperature = (AP::baro().get_temperature() + 40) * 10;
+        value.uint32 = pressure | (temperature << 19);
+        break;
+    }
+#endif
+
+#if AP_GPS_ENABLED
+    case IBUS_SENSOR_TYPE_GPS_LAT:
+        value.int32 = AP::gps().location().lat;
+        break;
+
+    case IBUS_SENSOR_TYPE_GPS_LNG:
+        value.int32 = AP::gps().location().lng;
+        break;
+
+    case IBUS_SENSOR_TYPE_GPS_ALT:
+        value.int32 = AP::gps().location().alt;
+        break;
+#endif // AP_GPS_ENABLED
+
+#if AP_BARO_ENABLED
+    case IBUS_SENSOR_TYPE_ALT:
+        value.int32 = AP::baro().get_altitude() * 100;
+        break;
+#endif
+
+    default:
+        value.int32 = 0;
+    }
+
+    return value;
+}
+
+uint16_t AP_IBus_Telem::get_average_cell_voltage_cV()
+{
+#if AP_BATTERY_ENABLED
```

Comment:
```suggestion
#if AP_BATTERY_ENABLED
uint16_t AP_IBus_Telem::get_average_cell_voltage_cV()
{
```

... and remove the call

Similarly elsewhere